hazelcast/src/main/java/com/hazelcast/internal/util/ThreadUtil.java

```diff
@@ -83,6 +88,24 @@ public static String createThreadPoolName(String hzName, String poolName) {
         return createThreadName(hzName, poolName) + ".thread-";
     }
 
+    /**
+     * Updates the thread name if matches pattern hz.hzName.poolName.thread-id to hz.hzName.targetPoolName.thread-id
+     *
+     * @param threadName     the name of the thread
+     * @param targetPoolName the target name of the pool
+     * @return the updated thread name if applicable, otherwise unchanged
+     * @throws java.lang.IllegalArgumentException if threadName or targetPoolName is null or empty
+     */
+    public static String updateThreadPoolName(String threadName, String targetPoolName) {
```

Comment:
Nitpicking, but I believe we can improve this further, here some concepts
- The name of the method implies that it changes the actual Thread name, rather than just updating a section of the String. Also the location of `ThreadUtil` enforces that feeling.
- I reckon its a bit complicated to rely on a Regex for the parsing, given that the input is a CSV (dot separated value), so you only have to replace one section.

So here is my proposal, we move the method inside `StringUtil` and call it something like `dsvReplaceSection(String dsv, int section, String replacement)` which I can imagine to be something like

```
    public static String dsvReplaceSection(String csv, String sep, int section, String replacement) {
        CharSequence[] sections = csv.split(sep);
        sections[section] = replacement;
        return String.join(".", sections);
    }
```

I believe the above is a strong utility function, with more chances to be used by other parts of the system too.

Reply:
I wanted to put this method in `ThreadUtil` since this is also the place that construct the original thread name. So that we can see the pattern originates from `ThreadUtil#createThreadPoolName`. The input thread name must match the regex before we attempt to change it. Once match, I am ok with either using `Matcher` or another `StringUtil` to update.

Would it make sense if I make the following adjustment?
* Add new `StringUtil` method to perform string replacement, for future reuse
* Rename current method in `ThreadUtil` to something else to indicate that current thread's name is not touched

Reply:
I am fine with the suggestion, but instead of renaming the method, why don't we set the name in that method?

Reply:
Updated. I keep csvReplaceSection simple and support only single character separator for now.